docs/QUICKSTART.md

```diff
@@ -0,0 +1,410 @@
+* git. It will be used to download the mbed codebase.
+* The mbed command-line tools, mbed-cli. You can run `pip install mbed-cli` to install them.
+
+For the remainder of this guide we will assume the following:
+
+* You are developing on a \*nix machine, in the `~/code` folder.
+* You are building the app for the [NXP FRDM-K64F](http://developer.mbed.org/platforms/FRDM-K64F/) target, with the [GCC ARM Embedded](https://launchpad.net/gcc-arm-embedded) toolchain.
+
+The instructions provided can be easily generalized to the case of other targets on other host OSs.
+
+## Start with the `blinky` app
+[Go to top](#overview)
+
+To create a new mbed application called `uvisor-example` just run the following commands:
+
+```bash
+$ cd ~/code
+$ mbed new uvisor-example
+```
+
+The mbed-cli tools will automatically fetch the mbed codebase for you. By default, git will be used to track your code changes, so your application will be ready to be pushed to a git server, if you want to.
+
+Once the import process is finished, create a `source` folder:
+```bash
+mkdir ~/code/uvisor-example/source
+```
+and place a new file `main.cpp` in it:
+
+```C
+/* ~/code/uvisor-example/source/main.cpp */
+
+#include "mbed.h"
+#include "rtos.h"
+
+DigitalOut led(LED1);
+
+int main(void)
+{
+    while (true) {
+        led = !led;
+        Thread::wait(500);
+    }
+}
+```
+
+This simple application just blinks an LED from the main thread, which is created by default by the OS.
+
+---
+
+**Checkpoint**
+
+Compile the application:
+
+```bash
+$ mbed compile -m K64F -t GCC_ARM
+```
+
+The resulting binary will be located at:
+
+```bash
+~/code/uvisor-example/.build/K64F/GCC_ARM/uvisor-example.bin
+```
+
+Drag-and-drop it onto the USB device mounted on your computer in order to flash the device. When the flashing process is completed, press the reset button on the device. You should see the device LED blinking.
+
+---
+
+In the next sections you will see:
+
+* How to [enable uVisor](#enable-uvisor) on the `uvisor-example` app.
+* How to [add a secure box](#add-a-secure-box) to the `uvisor-example` app with exclusive access to a timer, to a push-button interrupt, and to static and dynamic memories.
+
+## Enable uVisor
+[Go to top](#overview)
+
+To enable the uVisor on the app, just add the following lines at the beginning of the `main.cpp` file:
+
+```C
+/* ~/code/uvisor-example/source/main.cpp */
+
+#include "mbed.h"
+#include "rtos.h"
+#include "uvisor-lib/uvisor-lib.h"
+
+/* Register privleged system IRQ hooks.
+ * This is a system-wide configuration and it is independent from the app, but
+ * for the moment it needs to be specified in the app. This will change in a
+ * later version: The configuration will be provided by the OS. */
+extern "C" void SVC_Handler(void);
+extern "C" void PendSV_Handler(void);
+extern "C" void SysTick_Handler(void);
+UVISOR_SET_PRIV_SYS_IRQ_HOOKS(SVC_Handler, PendSV_Handler, SysTick_Handler);
+
+/* Main box ACLs. */
```

Comment:
This is the first mention of ACL in this document. It may be worth explicitly introducing the term someplace in this document, to keep the quick start self-contained (avoiding making the user read the entire README first).